analyze_test.go

```diff
@@ -1067,3 +1067,183 @@ func TestAnalyze_nap_app_protect_reconnect_period_seconds(t *testing.T) {
 		})
 	}
 }
+
+//nolint:funlen
+func TestAnalyze_http3(t *testing.T) {
+	t.Parallel()
+	testcases := map[string]struct {
+		stmt    *Directive
+		ctx     blockCtx
+		wantErr bool
+	}{
+		"http3 ok": {
+			&Directive{
+				Directive: "http3",
+				Args:      []string{"on"},
+				Line:      5,
+			},
+			blockCtx{"http", "server"},
+			false,
+		},
+		"http3 not ok": {
+			&Directive{
+				Directive: "http3",
+				Args:      []string{"somevalue"},
+				Line:      5,
+			},
+			blockCtx{"http", "server"},
+			true,
+		},
+		"http3_hq ok": {
+			&Directive{
+				Directive: "http3",
+				Args:      []string{"on"},
+				Line:      5,
+			},
+			blockCtx{"http", "server"},
+			false,
+		},
+		"http3_hq not ok": {
+			&Directive{
+				Directive: "http3",
+				Args:      []string{"somevalue"},
+				Line:      5,
+			},
+			blockCtx{"http", "server"},
+			true,
+		},
+		"http3_max_concurrent_streams ok": {
```

Comment:
nit, probably add not ok test cases with incorrect context` blockCtx{"http", "location"},`

Reply:
done